DataGenerator/DataGenerator.py

```diff
@@ -1,216 +1,125 @@
-            else:
-                datadict["Dose"] = np.expand_dims(ResampledDose, 0)
-
-            if self.transform:
-                try:
-                    transformed_data = self.transform(datadict["Dose"])
-                except:
-                    print(self.PatientList[id].label + ' has transform problem.')
-
-                if transformed_data is None:
-                    datadict["Dose"] = None
-                else:
-                    datadict["Dose"] = torch.from_numpy(transformed_data)
-        if 'PET' in self.keys:
-            PETScanPath = sorted(path.glob(patient_id + '*' + self.config['ImageSession']['PET'] + '*'))
-            PETScanPath = Path(PETScanPath[0], 'scans')
-            PET_match_folder = sorted(PETScanPath.glob('*-PET_AX_SC'))
-            if len(PET_match_folder) < 1:
-                raise ValueError(self.PatientList[id].label + ' should have one match!')
-            full_PET_path = Path(PET_match_folder[0], 'resources', 'DICOM', 'files')
-
-            PETreader = sitk.ImageSeriesReader()
-            dicom_names = sorted(PETreader.GetGDCMSeriesFileNames(str(full_PET_path)))
-            PETreader.SetFileNames(dicom_names)
-            PETObj = PETreader.Execute()
-            PET = sitk.GetArrayFromImage(CTObj)
-            if 'mask_name' in self.config['DATA'] and (mask_img is not None):
-                datadict["PET"] = np.expand_dims(MaskCrop(PET, cropbox), 0)
-            else:
-                datadict["PET"] = np.expand_dims(PET, 0)
-
-            if self.transform:
-                try:
-                    transformed_data = self.transform(datadict["PET"])
-                except:
-                    print(self.PatientList[id].label + 'has transform problem.')
-
-                if transformed_data is None:
-                    datadict["PET"] = None
-                else:
-                    datadict["PET"] = torch.from_numpy(transformed_data)
-
-            # print(datadict["Anatomy"].size, type(datadict["Anatomy"]))
-        if "Clinical" in self.keys:
-            category_feat, numerical_feat = LoadClinicalData(self.config, [self.PatientList[id]])
-            n_category_feat = self.c_norm.transform(category_feat).toarray()
-            n_numerical_feat = self.n_norm.transform(numerical_feat)
-            data = np.concatenate((n_numerical_feat, n_category_feat), axis=1)
-            data = np.squeeze(data)
-            # data = clinical_data.iloc[id].to_numpy()
-            # num_data = self.n_norm.transform([numerical_data.iloc[id]])
-            # cat_data = self.c_norm.transform([category_data.iloc[id]]).toarray()
-            # data = np.concatenate((num_data, cat_data), axis=1)
-            datadict["Clinical"] = data
-
-        if (self.inference):
-            return datadict
-        else:
-            return datadict, np.float32(label)
-
+            llabel = self.PatientList.loc[i,self.target]
+            label = torch.as_tensor(label, dtype=torch.int64)

```

Comment:
Not good if you have regression on the month, why cast the label as something else? Also line 89 you have the definiton of label, you might not need it there

Reply:
Changed in the new commits